chat-plugins/room-events.js

```diff
@@ -94,11 +94,85 @@ exports.commands = {
 		help: function (target, room, user) {
 			return this.parse('/help roomevents');
 		},
+		sortby: function (target, room, user) {
+			//preconditions
+			if (!room.chatRoomData) return this.errorReply("This command is unavailable in temporary rooms.");
+			if (!room.events || !Object.keys(room.events).length) {
+				return this.errorReply("There are currently no planned upcoming events for this room.");
+			}
+			if (!this.can('ban', null, room)) return false;
+
+			//declare variables
+			let multiplier = 1;
+			let columnName = "";
+			let delimited = target.split(target.includes('|') ? '|' : ',');
+			let sortable = Object.values(room.events);
+			let defaultCase = false;
+
+			//id tokens
+			if (delimited.length === 1) {
+				columnName = target;
+			} else {
+				let order = "";
+				[columnName, order] = delimited;
+				order = toId(order);
+				multiplier = (order === 'desc') ? -1 : 1;
+			}
+
+			//sort the array by the appropriate column name
+			columnName = toId(columnName);
+			switch (columnName) {
+			case "date":
+			case "eventdate":
+				sortable.sort((a, b) => { return (toId(a.date) < toId(b.date)) ? -1 * multiplier : (toId(b.date) < toId(a.date)) ? 1 * multiplier : 0; });
+				break;
+			case "desc":
+			case "description":
+			case "eventdescription":
+				sortable.sort((a, b) => { return (toId(a.desc) < toId(b.desc)) ? -1 * multiplier : (toId(b.desc) < toId(a.desc)) ? 1 * multiplier : 0; });
+				break;
+			case "eventname":
+			case "name":
+				sortable.sort((a, b) => { return (toId(a.eventName) < toId(b.eventName)) ? -1 * multiplier : (toId(b.eventName) < toId(a.eventName)) ? 1 * multiplier : 0; });
+				break;
+			default:
+				sortable.sort((a, b) => { return (toId(a.date) < toId(b.date)) ? -1 * multiplier : (toId(b.date) < toId(a.date)) ? 1 * multiplier : 0; });
+				defaultCase = true;
+			}
+
+			//rebuild the room.events object
+			room.events = {};
+			for (const sortedObj of sortable) {
+				const eventId = toId(sortedObj.eventName);
+				room.events[eventId] = sortedObj;
+			}
+			room.chatRoomData.events = room.events;
+
+			let resultString = "sorted list by column: ";
```

Comment:
From here to line 166 should be a lot shorter if you use conditionals (
``const resultString = `sorted by column: ${defaultCase ? "date (invalid column name provided)" : columnName} in` `` etc